tests/backup/backup_share_test.go

```diff
@@ -4215,3 +4215,177 @@ var _ = Describe("{SwapShareBackup}", func() {
+				for i := 0; i < numberOfBackups; i++ {
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					wg.Add(1)
+					go createBackup(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of user backups taken are: %v", userBackupMap)
+		})
+		Step(fmt.Sprintf("Share user %s backups with user %s with ViewOnlyAccess", user1, user2), func() {
+			log.InfoD(fmt.Sprintf("Sharing user %s backups with user %s with ViewOnlyAccess", user1, user2))
+			nonAdminCtx, err := backup.GetNonAdminCtx(user1, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", user1)
+			for backupName := range userBackupMap[user1] {
+				err := ShareBackup(backupName, nil, []string{user2}, ViewOnlyAccess, nonAdminCtx)
+				log.FailOnError(err, "failed to share user %s backup %s with user %s with ViewOnlyAccess", user1, backupName, user2)
+			}
+		})
+		Step(fmt.Sprintf("Delete the owner [%s] of the backups", user1), func() {
+			log.InfoD(fmt.Sprintf("Deleting the owner [%s] of the backups", user1))
+			err := backup.DeleteUser(user1)
+			log.FailOnError(err, "failed to delete user %s", user1)
+		})
+		Step(fmt.Sprintf("Delete user %s backups from the admin", user2), func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD(fmt.Sprintf("Deleting user %s backups from the admin", user2))
+			nonAdminCtx, err := backup.GetNonAdminCtx(user2, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", user2)
+			for backupName := range userBackupMap[user1] {
+				backupUid, err := Inst().Backup.GetBackupUID(nonAdminCtx, backupName, orgID)
+				log.FailOnError(err, "failed to fetch backup %s uid of the user %s", backupName, user2)
+				_, err = DeleteBackupWithClusterUID(backupName, backupUid, userClusterMap[user2][SourceClusterName], orgID, ctx)
+				log.FailOnError(err, "failed to delete backup %s of the user %s", backupName, user2)
+			}
+		})
+		Step(fmt.Sprintf("Wait for the backups to be deleted"), func() {
+			log.InfoD("Waiting for the backups to be deleted")
+			nonAdminCtx, err := backup.GetNonAdminCtx(user2, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", user2)
+			var wg sync.WaitGroup
+			for backupName := range userBackupMap[user1] {
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					err = Inst().Backup.WaitForBackupDeletion(nonAdminCtx, backupName, orgID, backupDeleteTimeout, backupDeleteRetryTime)
+					log.FailOnError(err, "failed while waiting for backup %s to be deleted", backupName)
+				}(backupName)
+			}
+			wg.Wait()
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		log.InfoD("Destroying the scheduled applications")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		DestroyApps(scheduledAppContexts, opts)
+		ctx, err := backup.GetAdminCtxFromSecret()
```

Comment:
Since we are passing admin context, Are we sure admin able to fetch user cluster during enumerate. Please cross-check here.

Reply:
Since there is only one user, this test case assumes that there will be only one source and one destination cluster. I cannot pass a non-admin context because the user has been deleted.